src/path.util.ts

```diff
@@ -1,35 +1,59 @@
 import { JsonSchema } from './models/jsonSchema';
 
-const toDataPathSegments = (path: string): Array<string> => {
-  const segments = path.split('/');
+/**
+ * Convert a schema path (i.e. JSON pointer) to an array by splitting
+ * at the '/' character and removing all schema-specific keywords.
+ *
+ * The returned value can be used to de-reference a root object by folding over it
+ * and derefercing the single segments to obtain a new object.
+ *
+ *
+ * @param {string} schemaPath the schema path to be converted
+ * @returns {string[]} an array containing only non-schema-specific segments
+ */
+const toDataPathSegments = (schemaPath: string): Array<string> => {
+  const segments = schemaPath.split('/');
   const startFromRoot = segments[0] === '#' || segments[0] === '';
   if (startFromRoot) {
     return segments.filter((segment, index) => {
       if (index === 0) {
         return false;
-      } else if (index % 2 === 1) {
-        return false;
       } else {
-        return true;
+        return index % 2 !== 1;
       }
     });
   }
-  return segments.filter((segment, index) => {
-    if (index % 2 === 0) {
-      return false;
-    } else {
-      return true;
-    }
-  });
+  return segments.filter((segment, index) => index % 2 !== 0);
 };
 
-export const toDataPath = (path: string): string => {
-  return toDataPathSegments(path).join('/');
+/**
+ * Remove all schema-specific keywords (e.g. 'properties') from a given path.
+ * @example
+ * toDataPath('#/properties/foo/properties/bar') === '#/foo/bar')
+ *
+ * @param {string} schemaPath the schema path to be converted
+ * @returns {string} the path without schema-specific keywords
+ */
+export const toDataPath = (schemaPath: string): string => {
+  return toDataPathSegments(schemaPath).join('/');
 };
 
-export const getValuePropertyPair = (instance: any, path: string):
+/**
+ * Resolve the given schema path against the given instance until the last
+ * segment. The returned value allows easy assignment of any new value.
+ *
+ * @example
+ * const pair = getValuePropertyPair(someData, someRef);
+ * pair.instance[pair.property] = someValue;
+ *
+ * @param {any} instance the instance to resolve the path against
+ * @param {string} schemaPath the schema path to be resolved
+ * @returns {{instance: string, property: string}} an object containing
```

Comment:
The double curly braces denote that an object is returned here, so this should be fine.